src/home/homework-cifs.c

```diff
@@ -28,7 +28,7 @@ int home_prepare_cifs(
                 char **pw;
                 int r;
 
-                r = home_unshare_and_mount(NULL, NULL, false);
+                r = home_unshare_and_mount(NULL, NULL, false, h->nosuid, h->nodev, h->noexec);
```

Comment:
hmm, please pass these as flags field, i.e. pass in user_record_mount_flags().

the simple directory backend, implemented in `src/home/homework-directory.c` actually did this right, it's the only backend currently using user_record_mount_flags() correctly for this. We should really do this the same way for the other backends too.